core/tauri/src/menu/icon.rs

```diff
@@ -207,7 +207,7 @@ impl<R: Runtime> IconMenuItem<R> {
   /// - **Windows / Linux**: Unsupported.
   pub fn set_native_icon(&mut self, _icon: Option<NativeIcon>) -> crate::Result<()> {
     #[cfg(target_os = "macos")]
-    return run_main_thread!(self, |mut self_: Self| self_
+    return run_main_thread!(self, |#[allow(unused_mut)] mut self_: Self| self_
```

Comment:
`set_native_icon` should not take `&mut self`, I fixed it in https://github.com/tauri-apps/tauri/pull/7709/files#diff-41ef79e158d69fe78154d145fa88e594ef8a3daf35e727db8ad9fa73cc23a663R211-R213, so either revert this or make the same changes to avoid conflicts in the future